Add query complexity reasonable use language (#3684)
fishtown-analytics/docs.getdbt.com: website/docs/docs/dbt-cloud-apis/discovery-querying.md

Every query will require an environment ID or job ID. You can get the ID from a dbt Cloud URL or using the Admin API.

> There are several illustrative example queries in this documentation. More examples are in the [use case guide](/docs/dbt-cloud-apis/discovery-use-cases-and-examples).

Reviewer: ```suggestion
There are several illustrative example queries on this page. For more examples, refer to [Use cases and examples for the Discovery API](/docs/dbt-cloud-apis/discovery-use-cases-and-examples).
```

## Reasonable use

Discovery (GraphQL) API usage is subject to request rate and response size limits to maintain the performance and stability of the metadata platform and prevent abuse.

> - The current request rate limit is 200 requests for a given IP address within a minute. Users who exceed this limit will receive an HTTP 429 response status.

Reviewer: ```suggestion
- The current request rate limit is 200 requests for a given IP address within a minute. If you exceed this limit, you will receive an HTTP 429 response status.
```

- Environment-level endpoints will be subject to response size limits in the future. The depth of the graph should not exceed three levels. A user can paginate up to 500 items per query.

> - Job-level endpoints are subject to query complexity limits. Nested nodes (e.g. parents), code (e.g. rawCode), and catalog columns count as most complex. Overly complex queries should be broken up into separate queries with only necessary fields included. The environment endpoint is recommended instead for most use cases to get the latest descriptive and result metadata for a dbt Cloud project.

Reviewer: ```suggestion
- Job-level endpoints are subject to query complexity limits. Nested nodes (like parents), code (like rawCode), and catalog columns are considered as most complex. Overly complex queries should be broken up into separate queries with only necessary fields included. dbt Labs recommends using the environment endpoint instead for most use cases to get the latest descriptive and result metadata for a dbt Cloud project.
```
